stetho/src/main/java/com/facebook/stetho/inspector/elements/android/FragmentDescriptor.java

```diff
@@ -64,6 +68,8 @@ protected int onGetChildCount(Object element) {
 
   @Override
   protected Object onGetChildAt(Object element, int index) {
+    verifyAccess();
```

Comment:
Over-zealous? No, I certainly disagree -- if this code doesn't execute on the UI thread, then super bad things can occur, and in subtle ways that aren't really deterministic. This over-zealousness has saved me on many, many occasions over the years in other UI code.

However, I think requiring every class to do this is a bit burdensome. I'm thinking that having something like AndroidDescriptor deriving from ChainedDescriptor and then it'll do the verifyAccess() before passing off to the derived class. At the very least it gives us a central place to implement this policy, rather than farming it out all over the place.